runtime/src/lib.rs

```diff
@@ -28,14 +28,43 @@ pub enum ClientError {
     WentAway = 0xFFFF_FE03,
 }
 
+/// Simple return type that is used when a function can only fail due to
+/// the server dying (i.e. it will never return an error code of its own)
+pub struct ServerDied;
+impl From<ServerDied> for u16 {
+    fn from(_: ServerDied) -> u16 {
+        0xDEAD
+    }
+}
+impl From<ServerDied> for u32 {
+    fn from(v: ServerDied) -> u32 {
+        u16::from(v) as u32
+    }
+}
+impl TryFrom<u32> for ServerDied {
+    type Error = ();
+    fn try_from(v: u32) -> Result<Self, Self::Error> {
+        match v {
+            0xDEAD => Ok(Self),
```

Comment:
This isn't right. Use `abi::extract_new_generation` to check whether the result is a dead code. The API isn't quite designed for this, so you'll need the slightly awkward `abi::extract_new_generation(v).is_some()`.

Reply:
Likewise, I just ended up with an empty `struct` with no `impl`